src/Diagnostics.Scripts/EntityInvoker.cs

```diff
@@ -329,14 +329,14 @@ private void Validate()
                 {
                     this._entryPointDefinitionAttribute.SupportTopicList.ToList().ForEach(item =>
                     {
-                        if (string.IsNullOrWhiteSpace(item.Id))
+                        if (string.IsNullOrWhiteSpace(item.Id) || string.IsNullOrWhiteSpace(item.SapSupportTopicId))
```

Comment:
Thinking about this change little bit:
- do we want to throw Script/Compile exception right now if the SAP Id is not present? This will block detector authors to publish their detectors next week.

How about this:
- In the beginning, We don't throw exceptions for this  SAP Id and let detector authors publish the detector changes
- When we have done the UI work and migration of support topic id to SAP Id in detector code, then we make this mandatory and throw exception on missing SAP Ids

Reply:
My intention here was to check at least **SupportTopicId** or **SapSupportTopicId** should exist. I put the check incorrectly but I was planning to do this so we don't block detector publish with author only specifying SapSupportTopicId :
`if (string.IsNullOrWhiteSpace(item.Id) && string.IsNullOrWhiteSpace(item.SapSupportTopicId))`

Reply:
@ShekharGupta1988 Do you think we should allow authors publishing detector with only SapSupportTopicId? Or we should discard checking for SAP Ids?

Reply:
Oh I see.. my bad. So basically one of the Id (either support topic id or SAp Id) should be present. Thats fine Cindy.
But the problem would be if some detector author provide just SAP Id and not support topic id. We will publish the detector but not surface it in case submission.

So until we do the whole wiring , its better to make support topic id as a mandatory field and we will change it once we migrate all detectors to SAP Id.

Reply:
Makes sense Shekhar! Will do the change!